Cannot save buffer when using built-int file picker

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

- Set `use_system_path_prompts` to false.
- Open a project.
- Open a project file.
- Run save as and type `./test.py` or `./src/test.py`.
- Observed the below error 

```
2024-07-29T12:57:01.923088Z [ERROR] invalid path "./test.py"

Stack backtrace:
   0: std::backtrace::Backtrace::create
   1: std::backtrace::Backtrace::capture
   2: <anyhow::Error>::msg::<alloc::string::String>
   3: anyhow::__private::format_err
   4: <worktree::Worktree>::write_file
   5: <project::buffer_store::BufferStore>::save_local_buffer
   6: <project::buffer_store::BufferStore>::save_buffer_as
   7: <project::Project>::save_buffer_as
   8: <editor::Editor as workspace::item::Item>::save_as
   9: <gpui::view::View<editor::Editor> as workspace::item::ItemHandle>::save_as
  10: <workspace::pane::Pane>::save_item::{closure#0}
  11: <workspace::Workspace>::save_active_item::{closure#0}::{closure#0}
  12: <async_task::raw::RawTask<<async_task::runnable::Builder<_>>::spawn_local::Checked<core::pin::Pin<alloc::boxed::Box<dyn core::future::future::Future<Output = core::result::Result<(), anyhow::Error>>>>>, core::result::Result<(), anyhow::Error>, <gpui::executor::ForegroundExecutor>::spawn::inner<core::result::Result<(), anyhow::Error>>::{closure#0}, ()>>::run
  13: <unknown>
  14: <unknown>
  15: <unknown>
  16: <unknown>
  17: <unknown>
  18: <unknown>
  19: <unknown>
  20: <unknown>
  21: <unknown>
  22: <unknown>
  23: <unknown>
  24: <unknown>
  25: <gpui::platform::mac::platform::MacPlatform as gpui::platform::Platform>::run
  26: <gpui::app::App>::run::<zed::main::{closure#2}>
  27: zed::main
  28: std::sys_common::backtrace::__rust_begin_short_backtrace::<fn(), ()>
  29: std::rt::lang_start::<()>::{closure#0}
  30: std::rt::lang_start_internal
  31: _main
  32: <unknown>

```

### Environment

Zed: v0.146.1 (Zed Preview)
OS: macOS 14.5.0
Memory: 32 GiB
Architecture: x86_64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>
<!-- Click below this line and paste or drag-and-drop your log-->

<!-- Click above this line and paste or drag-and-drop your log--></pre></details>

--- a/crates/file_finder/src/new_path_prompt.rs
+++ b/crates/file_finder/src/new_path_prompt.rs
@@ -248,7 +248,10 @@ impl PickerDelegate for NewPathDelegate {
         query: String,
         cx: &mut ViewContext<picker::Picker<Self>>,
     ) -> gpui::Task<()> {
-        let query = query.trim().trim_start_matches('/');
+        let query = query
+            .trim()
+            .trim_start_matches("./")
+            .trim_start_matches('/');
         let (dir, suffix) = if let Some(index) = query.rfind('/') {
             let suffix = if index + 1 < query.len() {
                 Some(query[index + 1..].to_string())

